src/qibolab/transpilers/abstract.py

```diff
@@ -0,0 +1,28 @@
+from abc import ABC, abstractmethod
+
+from qibo.models import Circuit
+
+
+class AbstractTranspiler(ABC):
```

Comment:
Just for the sake of drier names: you can avoid `Abstract`, since it is an `ABC` and it is inside `abstract.py` :)
```suggestion
class Transpiler(ABC):
```

There is no other object called `Transpiler`, and it makes sense that a list of them would be something like `list[Transpiler]`, since they won't be abstract any longer ^^